Python/ceval.c

```diff
@@ -5511,7 +5506,7 @@ call_trace(Py_tracefunc func, PyObject *obj,
     }
     else {
         initialize_trace_info(trace_info, frame);
-        frame->f_lineno = _PyCode_CheckLineNumber(frame->f_lasti, &trace_info->bounds);
+        frame->f_lineno = _PyCode_CheckLineNumber(frame->f_lasti*2, &trace_info->bounds);
```

Comment:
`_PyCode_CheckLineNumber` implies that it takes an instruction offset as its first argument, but in reality it's a bytecode offset. I suggest either changing `_PyCode_CheckLineNumber` to take a bytecode offset:

```diff
diff --git a/Objects/codeobject.c b/Objects/codeobject.c
index 69cb31c1ab..8fd65b48fc 100644
--- a/Objects/codeobject.c
+++ b/Objects/codeobject.c
@@ -1274,16 +1274,17 @@ _PyCode_InitAddressRange(PyCodeObject* co, PyCodeAddressRange *bounds)
 }
 
 /* Update *bounds to describe the first and one-past-the-last instructions in
-   the same line as lasti.  Return the number of that line, or -1 if lasti is out of bounds. */
+   the same line as the bytecode index, addrq.  Return the number of that line,
+   or -1 if lasti is out of bounds. */
 int
-_PyCode_CheckLineNumber(int lasti, PyCodeAddressRange *bounds)
+_PyCode_CheckLineNumber(int addrq, PyCodeAddressRange *bounds)
 {
-    while (bounds->ar_end <= lasti) {
+    while (bounds->ar_end <= addrq) {
         if (!PyLineTable_NextAddressRange(bounds)) {
             return -1;
         }
     }
-    while (bounds->ar_start > lasti) {
+    while (bounds->ar_start > addrq) {
         if (!PyLineTable_PreviousAddressRange(bounds)) {
             return -1;
         }

```

... or altering it to treat its first argument as an instruction offset (and update the callers):

```diff
diff --git a/Objects/codeobject.c b/Objects/codeobject.c
index 69cb31c1ab..41ace1c404 100644
--- a/Objects/codeobject.c
+++ b/Objects/codeobject.c
@@ -1278,12 +1278,13 @@ _PyCode_InitAddressRange(PyCodeObject* co, PyCodeAddressRange *bounds)
 int
 _PyCode_CheckLineNumber(int lasti, PyCodeAddressRange *bounds)
 {
-    while (bounds->ar_end <= lasti) {
+    int addrq = lasti * 2;  // Bytecode index.
+    while (bounds->ar_end <= addrq) {
         if (!PyLineTable_NextAddressRange(bounds)) {
             return -1;
         }
     }
-    while (bounds->ar_start > lasti) {
+    while (bounds->ar_start > addrq) {
         if (!PyLineTable_PreviousAddressRange(bounds)) {
             return -1;
         }

```